### PR TITLE
React to ANCM current addition

### PR DESCRIPTION
**Pull request by jkotalik**
Won't pass until ANCMV1 package is merged from https://github.com/aspnet/IISIntegration/pull/768

### CODE REVIEW

File: test/ServerComparison.TestSites/ServerComparison.TestSites.csproj [hunk 1]
```diff
@@ -5,9 +5,14 @@
     <RuntimeIdentifiers>win7-x86;win7-x64;linux-x64;osx-x64</RuntimeIdentifiers>
   </PropertyGroup>
 
-  <ItemGroup Condition="'$(OS)' == 'Windows_NT'">
+  <ItemGroup Condition="'$(OS)' == 'Windows_NT' AND ('$(ANCMVersion)' == 'V2' Or '$(ANCMVersion)' == '')">
     <PackageReference Include="Microsoft.AspNetCore.AspNetCoreModule" Version="$(MicrosoftAspNetCoreAspNetCoreModulePackageVersion)" />
```
Comment (Tratcher) [inline test/ServerComparison.TestSites/ServerComparison.TestSites.csproj]: So AspNetCoreModule is always the latest version, not necessarily the one we ship?

Comment (jkotalik) [reply test/ServerComparison.TestSites/ServerComparison.TestSites.csproj]: Potentially. We may want to remove the M.A.AspNetCoreModule nuget package and make it M.A.AspNetCoreModuleV2. I can make that change in a separate PR.

**Comment by Tratcher** [comment]
I think we need a new korebuild first.

**Comment by pakrym** [comment]
`run upgrade deps` or we need to take new CLI?

**Comment by jkotalik** [comment]
Still doesn't work. I'll investigate if we have a web.sdk with the change in it.

**Comment by jkotalik** [comment]
🆙 📅

File: test/ServerComparison.FunctionalTests/HelloWorldTest.cs [hunk 2]
```diff
@@ -33,14 +33,11 @@ public Task HelloWorld_HttpSys(ServerType serverType, RuntimeFlavor runtimeFlavo
         [ConditionalTheory]
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
-        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, HostingModel.InProcess)]
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: What happened to the rest of these?

Comment (jkotalik) [reply test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: Oops.

**Comment by jkotalik** [comment]
1000000% agree. We should consider how crazy these are when working on the deployers. These tests are effectively becoming an exploding matrix that are getting harder to read.

**Comment by Tratcher** [comment]
It's your test matrix that's exploding, the inline data is only a symptom. Refactoring the deployers isn't going to help you much. You could always try MemberData.

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 3]
```diff
@@ -33,12 +33,15 @@ public ResponseCompressionTests(ITestOutputHelper output) : base(output)
...
-        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone)]
-        public Task ResponseCompression_IISExpress_NoCompression(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, ApplicationType applicationType)
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Portable, "/p:ANCMVersion=V1", Skip = "Websdk issue with full framework publish. See https://github.com/aspnet/websdk/pull/322")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, "/p:ANCMVersion=V1")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, "/p:ANCMVersion=V2")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V1")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V2")]
+
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: extra line?

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 4]
```diff
@@ -67,20 +70,24 @@ public Task ResponseCompression_Nginx_NoCompression(ServerType serverType, Runti
...
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task ResponseCompression_IISExpress_HostCompression_CLR()
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V1")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V2")]
+        public Task ResponseCompression_IISExpress_HostCompression_CLR(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, ApplicationType applicationType, string additionalPublishParameters)
         {
-            return ResponseCompression(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, CheckHostCompressionAsync, ApplicationType.Portable, hostCompression: true);
+            return ResponseCompression(serverType, runtimeFlavor, architecture, CheckHostCompressionAsync, applicationType, hostCompression: true, additionalPublishParameters: additionalPublishParameters);
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: Why did you move these out?

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 5]
```diff
@@ -92,12 +99,14 @@ public Task ResponseCompression_Nginx_HostCompression(ServerType serverType, Run
...
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task ResponseCompression_Windows_AppCompression_CLR()
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V1")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Standalone, "/p:ANCMVersion=V2")]
+        public Task ResponseCompression_Windows_AppCompression_CLR(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, ApplicationType applicationType, string additionalPublishParameters)
         {
-            return ResponseCompression(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x64, CheckAppCompressionAsync, ApplicationType.Portable, hostCompression: false);
+            return ResponseCompression(serverType, runtimeFlavor, architecture, CheckAppCompressionAsync, applicationType, hostCompression: false, additionalPublishParameters: additionalPublishParameters);
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: This was WebListener, not IISExpress

File: test/ServerComparison.FunctionalTests/ResponseTests.cs [hunk 6]
```diff
@@ -125,6 +125,21 @@ public Task ResponseFormats_Windows_ManuallyChunk(ServerType serverType, Runtime
...
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, HostingModel.OutOfProcess, "/p:ANCMVersion=V2")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, HostingModel.OutOfProcess, "/p:ANCMVersion=V2")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, HostingModel.InProcess, "/p:ANCMVersion=V2")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, HostingModel.InProcess, "/p:ANCMVersion=V2")]
+
+        public Task ResponseFormats_IIS_ManuallyChunk(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, ApplicationType applicationType, HostingModel hostingModel, string additionalPublishParameters)
+        {
+            return ResponseFormats(serverType, runtimeFlavor, architecture, CheckManuallyChunkedAsync, applicationType, hostingModel: hostingModel, additionalPublishParameters: additionalPublishParameters);
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseTests.cs]: Embed any consistent parameters to make the InlineData more legible.

**Comment by jkotalik** [comment]
🆙 📅 to organize tests better. Also, https://github.com/aspnet/ServerTests/pull/109/files#diff-ccc0d19f30eb63b18f8608680bc92b21R57 should fix the current failures here: http://aspnetci/viewType.html?buildTypeId=Lite_MusicStoreTest

File: test/ServerComparison.FunctionalTests/HelloWorldTest.cs [hunk 7]
```diff
@@ -20,32 +20,31 @@ public HelloWorldTests(ITestOutputHelper output) : base(output)
...
-        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, HostingModel.OutOfProcess)]
-        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, HostingModel.OutOfProcess)]
-        [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Portable, HostingModel.OutOfProcess, Skip = "Websdk issue with full framework publish. See https://github.com/aspnet/websdk/pull/322")]
-        public Task HelloWorld_IISExpress(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, ApplicationType applicationType, HostingModel hostingModel)
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Standalone, HostingModel.OutOfProcess, "/p:ANCMVersion=V1")]
+        [InlineData(ServerType.IISExpress, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, ApplicationType.Portable, HostingModel.OutOfProcess, "/p:ANCMVersion=V1")]
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/HelloWorldTest.cs]: I thought you were going to remove ServerType from the InlineData?

File: test/ServerComparison.FunctionalTests/ResponseTests.cs [hunk 8]
```diff
@@ -24,52 +24,121 @@ public ResponseTests(ITestOutputHelper output) : base(output)
...
+        {
+            return ResponseFormats(ServerType.WebListener, runtimeFlavor, RuntimeArchitecture.x64, CheckManuallyChunkedAsync, applicationType);
+        }
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [InlineData(RuntimeFlavor.Clr, ApplicationType.Portable)]
+        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable)]
+        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone)]
+        // [InlineData(ServerType.IISExpress, RuntimeFlavor.Clr, RuntimeArchitecture.x64, ApplicationType.Portable)] // https://github.com/aspnet/IISIntegration/issues/7
```
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseTests.cs]: Delete
